docs/src/developer/reference/oauth.md

```diff
@@ -0,0 +1,402 @@
+| `state`         | Required. An opaque value used by the client to maintain state between the request and callback.</br>The authorization server includes this value when redirecting the user-agent back to the client.</br>The parameter SHOULD be used for preventing cross-site request forgery. |
+
+Once the user consents the user agent will be redirected back to the 3rd party app, using the redirect URI provided during client registration, with an authorization code which can be used to retrieve an access token and a refresh token and is good for one use.
+
+Example response:
+
+```http
+HTTP/1.1 302 Found
+Transfer-Encoding: chunked
+Date: Thu, 23 Feb 2023 15:50:21 GMT
+Server: Warp/3.3.23
+Location: https://example.com?code=1395a1a44b72e0b81ec8fe6c791d2d3f22bc1c4df96857a88c3e2914bb687b7b&state=foobar
+Vary: Accept-Encoding
+```
+
+### Retrieve access and refresh token
+
+The 3rd party app sends the authorization code together with the client credentials and the parameters shown below using the `application/x-www-form-urlencoded` format with character encoding of UTF-8 to the authorization server to retrieve an access token and a refresh token:
+
+```curl
+curl -s -X POST localhost:8080/oauth/token \
+  -H "Content-Type: application/x-www-form-urlencoded" \
+  -d 'code=1395a1a44b72e0b81ec8fe6c791d2d3f22bc1c4df96857a88c3e2914bb687b7b&client_id=b9e65569-aa61-462d-915d-94c8d6ef17a7&grant_type=authorization_code&redirect_uri=https%3A%2F%2Fexample.com&client_secret=3f6fbd62835859b2bac411b2a2a2a54699ec56504ee32099748de3a762d41a2d'
+```
+
+Parameters:
+
+| Parameter       | Description                                                                             |
+| --------------- | --------------------------------------------------------------------------------------- |
+| `code`          | Required. The authorization code received from the authorization server.                |
+| `client_id`     | Required. The client identifier.                                                        |
+| `grant_type`    | Required. Value MUST be set to `authorization_code`.                                    |
+| `redirect_uri`  | Required. The value MUST be identical to the one provided in the  authorization request |
+| `client_secret` | Required. The client's secret.                                                          |
+
+Example response:
+
+```JSON
+{
+  "accessToken": "eyJhbGciOiJFZERTQSJ9.eyJhdWQiOiJleGFtcGxlLmNvbSIsImV4cCI6MS42NzcyMzYyNDkxMTIzMTk3MWU5LCJpYXQiOjEuNjc3MjM2MjQ2MTEyMzE5NzFlOSwiaXNzIjoiZXhhbXBsZS5jb20iLCJzY29wZSI6InJlYWQ6c2VsZiIsInN1YiI6ImJhOTIxY2ZmLWU1ZWEtNDMxNS1iZTNkLWZiNjA3NTU0M2Y3MCJ9.5ksjS7msi9NSNat-qh7-Y5O-u9TcuYeLWTAsiAyes_oLwfjD_jYtGevUAuiVV6RXgPBO00VEMv-ZS86e7sd5Dg",
+  "expiresIn": 300,
+  "refreshToken": "eyJhbGciOiJFZERTQSJ9.eyJzdWIiOiI4NzI1ZTRkNC01Njc5LTQwZGEtOTI3My03YTBkMmIwYjUwMGYifQ.59IICzGoli5nfwJ1ZwRH_b3T-lRgBrralE1EZZRtadI2eKrta0kaLIZpuMWPC2Icj6-LSEBsyYLXpxOm3cNaDw",
+  "tokenType": "Bearer"
```

Comment:
I'm not sure I like having an example here. OAUTH itself says these tokens are opaque, but this seems to be a JWT - even though there's no mention of the endpoints specified by [RFC8414](https://www.rfc-editor.org/rfc/rfc8414.html).

Also, are you sure about the key names here? Section 4.1.4 in [RFC 6749](https://www.ietf.org/rfc/rfc6749.txt) uses snake case.

Reply:
I changed it to snake case.